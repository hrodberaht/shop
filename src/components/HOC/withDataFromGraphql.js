import React, { Component } from 'react';

const withDataFromGraphql = (WrappedComponent) => {
  class WithDataFromGraphql extends Component {
    render() {
      if (this.props.fetchedData.loading) return <p>Loading...</p>;
      if (this.props.fetchedData.error) return <p>Error :(</p>;
      return <WrappedComponent {...this.props} />;
    }
  }

  return WithDataFromGraphql;
};

export default withDataFromGraphql;
