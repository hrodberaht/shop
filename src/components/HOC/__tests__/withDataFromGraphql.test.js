import React from 'react';
import withDataFromGraphql from '../withDataFromGraphql';

const defaultProps = {
  fetchedData: {
    data: {
      orders: [{}, {}],
    },
    loading: true,
    error: null,
  },
};

const mockComponent = () => <p>Fake component</p>;

const componentWithHOC = withDataFromGraphql(mockComponent);

const setup = buildSetup(componentWithHOC, defaultProps);

describe('<withDataFromGrapql />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render mockComponent if loading false', () => {
    const {
      wrapper,
      props: { fetchedData },
    } = setup();
    wrapper.setProps({ fetchedData: { ...fetchedData, loading: false } });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error if error and loading false', () => {
    const {
      wrapper,
      props: { fetchedData },
    } = setup();
    wrapper.setProps({ fetchedData: { ...fetchedData, loading: false, error: 'error' } });
    expect(wrapper).toMatchSnapshot();
  });
});
