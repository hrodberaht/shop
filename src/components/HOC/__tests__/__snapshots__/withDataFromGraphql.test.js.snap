// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withDataFromGrapql /> should render error if error and loading false 1`] = `
<p>
  Error :(
</p>
`;

exports[`<withDataFromGrapql /> should render mockComponent if loading false 1`] = `
<mockComponent
  fetchedData={
    Object {
      "data": Object {
        "orders": Array [
          Object {},
          Object {},
        ],
      },
      "error": null,
      "loading": false,
    }
  }
/>
`;

exports[`<withDataFromGrapql /> should render without crash 1`] = `
<p>
  Loading...
</p>
`;
